openvdb_points/tools/AttributeArray.h

```diff
@@ -1562,18 +1563,61 @@ TypedAttributeArray<ValueType_, Codec_>::isEqual(const AttributeArray& other) co
 
 ////////////////////////////////////////
 
+
+/// Accessor to call unsafe get and set methods based on templated Codec and Value
+template <typename CodecType, typename ValueType>
+struct AccessorEval
+{
+    typedef ValueType (*GetterPtr)(const AttributeArray* array, const Index n);
+    typedef void (*SetterPtr)(AttributeArray* array, const Index n, const ValueType& value);
+
+    /// Getter that calls to TypedAttributeArray::getUnsafe()
+    /// @note Functor argument is provided but not required for the generic case
+    static ValueType get(GetterPtr /*functor*/, const AttributeArray* array, const Index n) {
+        return TypedAttributeArray<ValueType, CodecType>::getUnsafe(array, n);
+    }
+
+    /// Getter that calls to TypedAttributeArray::setUnsafe()
+    /// @note Functor argument is provided but not required for the generic case
+    static void set(SetterPtr /*functor*/, AttributeArray* array, const Index n, const ValueType& value) {
+        TypedAttributeArray<ValueType, CodecType>::setUnsafe(array, n, value);
+    }
+};
+
+
+/// Partial specialization when Codec is not known at compile-time to use the supplied functor instead
+template <typename ValueType>
+struct AccessorEval<UnknownCodec, ValueType>
+{
+    typedef ValueType (*GetterPtr)(const AttributeArray* array, const Index n);
+    typedef void (*SetterPtr)(AttributeArray* array, const Index n, const ValueType& value);
+
+    /// Getter that calls the supplied functor
+    static ValueType get(GetterPtr functor, const AttributeArray* array, const Index n) {
+        return (*functor)(array, n);
+    }
+
+    /// Setter that calls the supplied functor
+    static void set(SetterPtr functor, AttributeArray* array, const Index n, const ValueType& value) {
+        (*functor)(array, n, value);
+    }
+};
+
+
+////////////////////////////////////////
+
```

Comment:
We could maybe sue boost primitives so we don't have to have this class?

```
template<typename ValueType, typename Codec = UnknownCodec>
struct Handle
{
	ValueType get() {
		return get<Codec>();
	}

private:

	template<typename __Codec>
	typename boost::enable_if_c<boost::is_same<__Codec, UnknownCodec>::value, ValueType>::type
	get() {
		 return (*functor)(array, n);
	}

	template<typename __Codec>
	typename boost::enable_if_c<!boost::is_same<__Codec, UnknownCodec>::value, ValueType>::type
	get() {
		 return TypedAttributeArray<ValueType, __Codec>::getUnsafe(array, n);
	}
};

```

Unfortunately needs a second private get method as the get method itself is not templated (sfinae on enable_if). Potentially wouldn't need this in c++0x with default function templates?